tests/test_node_ci_modernizer.py

```diff
@@ -0,0 +1,59 @@
+"""
+Tests for Github Actions Modernizer Script
+"""
+import os
+import shutil
+import uuid
+from unittest import TestCase
+
+from edx_repo_tools.codemods.node16 import GithubCINodeModernizer
+from edx_repo_tools.utils import YamlLoader
+
+
+class TestGithubActionsModernizer(TestCase):
+
+    def setUp(self):
+        self.test_file1 = self._setup_local_copy("sample_files/sample_node_ci.yml")     # file without strategy matrix
+        self.test_file2 = self._setup_local_copy("sample_files/sample_node_ci2.yml")    # file with strategy matrix
+
+        self.ci_elements1 = TestGithubActionsModernizer._get_updated_yaml_elements(self.test_file1)
+        self.ci_elements2 = TestGithubActionsModernizer._get_updated_yaml_elements(self.test_file2)
+
+    @staticmethod
+    def _setup_local_copy(file_name):
+        current_directory = os.path.dirname(__file__)
+        temp_file = os.path.join(current_directory, str(uuid.uuid4()) + ".yml")
+        local_file = os.path.join(current_directory, file_name)
+        shutil.copy2(local_file, temp_file)
+        return temp_file
+
+    @staticmethod
+    def _get_updated_yaml_elements(file_path):
+        modernizer = GithubCINodeModernizer(file_path)
+        modernizer.modernize()
+        yaml_loader = YamlLoader(file_path)
+        return yaml_loader.elements
+
+    def test_matrix_items(self):
+        node1 = self.ci_elements1['jobs']['tests']['strategy']['matrix']['node']
+        node2 = self.ci_elements2['jobs']['tests']['strategy']['matrix']['node']
+
+        self.assertIn(16, node1)
+        self.assertIn(16, node2)
+
+    def test_node_version_value(self):
+        for step in self.ci_elements1['jobs']['tests']['steps']:
+            if step['name'] == 'Setup Nodejs':
+                self.assertEqual(step['with']['node-version'], '${{ matrix.node }}')
```

Comment:
instead of doing `assert` inside the loop we should filter the steps first and then do an `assert`